lightning/src/ln/chan_utils.rs

```diff
@@ -550,6 +556,415 @@ pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, contest_del
 	}
 }
 
+/// Static channel fields used to build transactions given per-commitment fields (CommitmentTransactionInfo)
+#[derive(Clone)]
+pub struct ChannelStaticInfo {
+	/// Holder public keys
+	pub holder_pubkeys: ChannelPublicKeys,
+	/// Counter-party public keys
+	pub counterparty_pubkeys: ChannelPublicKeys,
+	/// The contest delay selected by the holder
+	pub holder_selected_contest_delay: u16,
+	/// The contest delay selected by the counterparty
+	pub counterparty_selected_contest_delay: u16,
+	/// The funding outpoint
+	pub funding_outpoint: OutPoint,
+	/// Whether the channel is outbound from the point of view of the holder
+	pub is_outbound_from_holder: bool,
+}
+
+impl_writeable!(ChannelStaticInfo, 0, {
+	holder_pubkeys,
+	counterparty_pubkeys,
+	holder_selected_contest_delay,
+	counterparty_selected_contest_delay,
+	funding_outpoint,
+	is_outbound_from_holder
+});
+
+impl ChannelStaticInfo {
+	/// Get the pubkeys for the selected side and the opposing side
+	pub fn pubkeys(&self, holder: bool) -> (&ChannelPublicKeys, &ChannelPublicKeys) {
+		if holder {
+			(&self.holder_pubkeys, &self.counterparty_pubkeys)
+		} else {
+			(&self.counterparty_pubkeys, &self.holder_pubkeys)
+		}
+	}
+
+	/// Get the contest delay applicable to the selected side's transactions.
+	/// Note that the contest delay was selected by the opposite party.
+	pub fn contest_delay(&self, holder: bool) -> u16 {
+		if holder { self.counterparty_selected_contest_delay } else { self.holder_selected_contest_delay }
+	}
+
+	/// Whether the channel is outbound from the perspective of the selected party
+	pub fn is_outbound(&self, holder: bool) -> bool {
+		return if holder { self.is_outbound_from_holder } else { !self.is_outbound_from_holder };
+	}
+}
+
+/// This class tracks the information needed to build a holder's commitment transaction and to actually
+/// build and sign.  It is used for holder transactions that we sign only when needed.
+#[derive(Clone)]
+pub struct HolderCommitmentTransactionInfo {
```

Comment:
As a name suggestion what about `HolderRawChannelTxn`, what we want to convey is a) this is scoping both commitment and htlc txn b) this is "raw" in opposition to signer finalized.

Reply:
I renamed this to `HolderCommitmentTransaction` (and renamed the old one to something else).  I feel that "Raw" isn't that descriptive.

Let me know what you think.

Reply:
I like dropping the `Info` on new names. 

I would like to convey better the different commitment package states : channel_parameters (step 1) --> channel_parameters + CommitmentTransaction (dynamic) (step 2) --> channel_parameters + CommitmentTransaction + counterparty_sigs (step 3) --> channel_parameters + CommitmentTransaction + counterparty_sigs + holder_sigs (step 4).

Note that you have a bifurcation at (step 2) if it's for counterparty, you will produce the holder sigs and send them through `commitment_signed`. Otherwise, if it's for holder, you will pass the state to our chain monitor. 

That's the gross graph I hope we move towards to make it clearer for future reviewers, but not need to stick to it rn.